llvm/include/llvm/IR/LLVMContext.h

```diff
@@ -335,6 +335,14 @@ class LLVMContext {
   StringRef getDefaultTargetFeatures();
   void setDefaultTargetFeatures(StringRef Features);
 
+  /// Key Instructions: update the highest number atom group emitted for any
+  /// function.
+  void updateAtomGroupWaterline(uint64_t G);
+
+  /// Key Instructions: get the next free atom group number and increment
+  /// the global tracker.
+  uint64_t incNextAtomGroup();
+
```

Comment:
IMO in isolation it's not clear that this is to do with debugging information and source locations; could we shoe-horn `DILocation` into the comments to make it clear what it affects?

(Thinking purely about someone stumbling on this and not immediately knowing whether it's relevant to what they're studying)